pkg/validators/version/served/served_test.go

```diff
@@ -0,0 +1,541 @@
+// Copyright 2025 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package served
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/crdify/pkg/config"
+	"sigs.k8s.io/crdify/pkg/validations"
+	"sigs.k8s.io/crdify/pkg/validations/property"
+)
+
+type createCRDOption func(crd *apiextensionsv1.CustomResourceDefinition)
+
+func createCRD(options ...createCRDOption) *apiextensionsv1.CustomResourceDefinition {
```

Comment:
I'm guessing these CRD generation helpers are either duplicate of some other helpers in this repo (or they would be helpful elsewhere).

Just calling that out in case you'd rather have these elsewhere, or if there's something I can reuse.